### ISSUE TITLE
Error adding multiple filters (seems to start when adding the 4th filter)

### ISSUE THREAD

**Issue by gdneye** [licoffe/POE-sniper#2]
I've started using the app and I'm having a problem every time I try to add a 4th filter.  I've tried different filters and it errors each time so I don't think its an issue resulting from the specific filter I'm using.

I've uploaded a video showing the issue to youtube...

https://youtu.be/ws1RjKRDn4o

In the video you can see that my 3rd filter is a Goldwyrm and is added fine, but my 4th filter (an eye of Chayula) appeard to load, but if you look at the affixes for both the goldwyrm and the eye you can see that both of them adopt each other's affixes.  Its not in the video but the price for Goldwyrm is also updated to the amount I listed for the eye. 

Attempting to edit either filter (eye or Goldwyrm) removes the other filter from the filter list (although the filter list still lists 4 total filters).  Deleting the remaining filter will reduce the filter list from 4 filters to 2 filters.  All of this suggests to me that the two filters somehow become combined.

**Comment by gdneye** [comment]
Figured out how to capture my full screen so let me know if you need me to redo the video so you can see all of the process.

**Comment by licoffe** [comment]
Thank you for reporting the issue. I've ran into this issue several times on my side as well. I'll report back to you when it's fixed.

**Comment by gdneye** [comment]
Sounds good, thanks!

**Comment by licoffe** [comment]
I'm in the process of splitting the code into several modules instead of the current monolithic form. It seems it also fixed the issue you were encountering in the process. I'll keep you posted on how it turns out. I'm not committing just yet since splitting the code broke some of the interaction (filter removal for instance). I need to bind the interface elements with the new modules and I'll be able to add new features. Btw, out of curiosity, how did you learn about this project?

**Comment by gdneye** [comment]
**TL:DR I've been working on/looking for an app that would do this for a few months.  While modifying a less complete project I'd previously found I realized I was coming up with additional features faster than I was implementing the ones I'd already thought of.  I went searching for a better sniper tool (github is my primary source for poe tools) and found yours.  DL'ed and ran it and saw that it has solved most of the features I was working on.**

Somewhat longer version...  I started off working on building my own sniper (but I'm not a programmer).  I built a working program in JS (a friend/business partner of mine who is a developer helped me with the basics and helped me get over a couple of hurdles I couldn't solve with Google/W3 Schools).  I've written VBA and HTML code before, have a good understanding of SQL and I have been able to read (but not write) his JS well enough to trouble shoot his code.  

I ended up building a functional program with almost no QoL features.  It was very useful in terms of gaining an understanding of how the river works, etc,  but near the end of building it I ran across this project... 

https://github.com/Tonexus/poe-scambot

Tonexus' interface had a number of features I was hoping to add to my project (such as making a sound, having a better search function and auto-copying to the clipboard).  Since his was definitely farther along than mine I started working on modifying his code and abandoned mine.  I made some pretty good progress, and have his project at a point where I can input my Regex queries directly into his code.  This allows me to search multiple, differently priced queries at once.  (I had to knock out a few parts of his "parserthread" and add price to his "full_text" variable so I could do individual name+properties+price searches and made a few other modifications, but that part of it is currently working they way I wanted.)

Still, this required me to manually input and maintain a large number of searched items and prices as well as relative currency values that were likely to change during and between leagues (your UI would allow simple adjustments when I notice a problem).  His app also does not handle currency conversion, which I cannot do in Python without a fair bit of learning.  I started trying to think through how I could solve that with my current abilities.  I was contemplating pulling currency values into excel from poe.ninja and building an excel file to auto-generate the regex needed.  I'm pretty confident I could do that with VBA, but... I was spending more time thinking about new features than I was actually using the app.  I realized I should just see if there was a better project out there (Tonexus has also stopped updating his project, so feature requests wasn't really an option).

Github had originally led me to Tonexus' program, and a few days ago when I started realizing how much modification I was going to need I looked around github again.  Yours is the most complete sniper app I've found to date (and you are still actively developing it).

That may have been more info that you were interested in =), but that is how I found your project.  

I'm pretty excited about it though, I've been working on solving this problem since about the beginning of Legacy League and your project is quite a bit farther than any other project I've found (or the progress I've made).  Trading is a huge part of what I enjoy about POE, but like I said, 70 windows gets out of hand and can still be pretty limiting.  A well built app could really take that to the next level.

**Comment by licoffe** [comment]
Wow, thanks for the whole story :) 

Yeah, as you said there are many parts to get covered:
- Currency rates
- API download
- Interface
- Testing
- ...

> His app also does not handle currency conversion, which I cannot do in Python without a fair bit of learning. I started trying to think through how I could solve that with my current abilities. I was contemplating pulling currency values into excel from poe.ninja and building an excel file to auto-generate the regex needed. I'm pretty confident I could do that with VBA, but...

I wrote some code to compute stats on item prices (median, mode, average, min, max), including currencies a while back and I released a website [(poe-rates.com)](http://poe-rates.com) at the beginning of the Legacy league to help players price their items. I'm using the currency rates in chaos from this platform now.

> 70 windows gets out of hand and can still be pretty limiting. A well built app could really take that to the next level.

I really hope my app can handle 70 filters but I have honestly no idea how much computing power it would use since I haven't tried it yet. I may have to do some optimizations in the code here and there :D

**Comment by licoffe** [comment]
I published a new release. Could you try and tell me if it fixes your issue?

**Comment by gdneye** [comment]
Downloaded it and began testing.  The first time I tried it out I started running into a similar issue, but I closed and re-loaded the program and it has been working fine so far.  I think my issue the first time was that I was attempting to do things too quickly.  Specifically, if after hitting "Add Filter" I try to do something else quickly it would cause an error.  If I wait until the new filter fully loads it seems fine (I have found that once the poe.trade stats in teal are loaded the process seems to have completed and I am safe to start working on my next filter.  It's not bothering me, so it's really just fyi.  I'll keep going and see if I run into any further issues.

**Comment by licoffe** [comment]
It's possible that your filters were also corrupted with the first version.
